### PR TITLE
PHP 8.2 support

### PR DESCRIPTION
**Pull request by patrickomeara**
This PR updates `voku/portable-utf8` where a string interpolation deprecation has been fixed.

It also updates the test files to match the iterator parent return types. This upgrade means that the tests will only run on PHP 8.0 and above, due to `mixed` not being available on earlier versions. However, this matches the official supported versions https://www.php.net/supported-versions.php

image

I've preemptively updated the README assuming there will be a new major version released.

I've also added a github actions workflow to test PHP 8.0, 8.1 and 8.2 but Actions will need to be turned [on for this repo](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository). There may need to be changes to the file after it has been ran. I'm just basing it off other files across public and my private repos.

### CODE REVIEW

**Comment by SharkyKZ** [comment]
What's the practical reason for increasing PHP requirements instead of using attributes?

**Comment by patrickomeara** [comment]
Hi @SharkyKZ thanks for your question, the practical reason is so that this package works with PHP 8.2 out of the box, which it currently does not.

As the README states there are three major versions of this package, all targeting specific PHP versions. This follows semantic versioning, where breaking changes get a new major.

https://github.com/wamania/php-stemmer/blob/1601ca7f3c34be80bad8c23ac55954658b688888/README.md?plain=1#L30-L46

Attributes are a possibility, I'll let the maintainers decide which approach they would like to take. Looking forward to moving this package forward.

**Comment by SharkyKZ** [comment]
That is not a practical reason because narrowing PHP version support is not required to support PHP 8.2. It's a change for the sake of change. Of course, it's up to maintainers to decide. But with current code base there is no gain. Even the hard v6 requirement for `voku/portable-utf8` is technically not necessary. Just my 2c.

**Comment by patrickomeara** [comment]
It's not change for the sake of change, out of the box this package is causing errors when running Psalm on PHP 8.2.

```
./vendor/bin/psalm
Uncaught RuntimeException: PHP Error: Using ${var} in strings is deprecated, use {$var} instead in /Users/pat/Code/web/vendor/voku/portable-utf8/src/voku/helper/UTF8.php:3880 in /Users/pat/Code/web/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
```

`voku/portable-utf8` has recognised this and released a fix in 6.0.5 https://github.com/voku/portable-utf8/pull/181

Not pulling that updated package in because it's "technically not necessary" is not moving this package forward.

**Comment by SharkyKZ** [comment]
I think you misunderstood. Supporting v6 is fine but dropping v5.4 was technically unnecessary. Likewise, there was no practical reason to drop PHP 7.3 support when you can simply use `ReturnTypeWIllChange` attribute to suppress the warnings. The narrowing of supported version is, in fact, a change for the sake of change in this case.

**Comment by wamania** [comment]
Hi
I use your github actions file directly, it's something that should have already been there. Now we have ci tests.

I don't really want to have too many versions. It's possible to keep the compatibility with 7.3 and upgrade portable-utf8 (to be compatible php8.2)
As it's your PR, can you modify it to keep the compatibility with 7.3 and only add support for php8.2
there is no need here to have breaking changes.

Thanks !

**Comment by patrickomeara** [comment]
I've added PHP 8.2 to the test matrix, and added back in the stability of lowest and stable 

All test are now passing https://github.com/patrickomeara/php-stemmer/actions/runs/4129148443

**Comment by Namoshek** [comment]
As a workaround, since this library requires no changes when using `voku/portable-utf8:^6.0`, use an inline alias for the version in your `composer.json`:
```json
"require": {
    "voku/portable-utf8": "6.0.12 as 5.4.0"
}
```

**Review by HLeithner — APPROVED** [review]
Looks good to me

**Comment by HLeithner** [comment]
@wamania can we merge this and release a new version please?

**Review by ChristophWurst — APPROVED** [review]
Looks good

**Pull request merged**
